### DISCUSSION TITLE
Query parameters for view collections

### DISCUSSION THREAD

**Discussion by slimcdk** [pocketbase/pocketbase#2687] [Q&A]
Is it possible to provide parameters for the query on view collections? E.i. like:
```
SELECT
  (ROW_NUMBER() OVER()) as id,
  table1.alias AS table1_alias,
  table1.id AS table1_id,
  table2.name AS table2_name,
  table2.id AS table2_id
FROM guinea_pigs
LEFT JOIN table1 ON table2.table1_id = table1.id
WHERE table1.alias LIKE '%{query-parameter}%'
```

**Comment by dschissler** [comment]
You need to use an API Filter.  I'm not sure if this is exactly correct.

`@collection.table1.alias  ~ @request.query.MyParam`

[Basics - API rules and filters](https://pocketbase.io/docs/api-rules-and-filters/)

API Filters will add SQL WHERE terms to the Query of the view collection.
